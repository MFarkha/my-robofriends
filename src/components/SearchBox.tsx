import { ChangeEventHandler } from "react";

type SearchBoxProps = {
    searchChange: ChangeEventHandler<HTMLInputElement>
};

const SearchBox = ({ searchChange }: SearchBoxProps) => {
    return (
        <div className="pa2">
            <input 
                aria-label="search robots"
                className="pa3 ba b--green bg-lightest-blue" 
                type='search' 
                name="searchBox"
                placeholder="search robots"
                onChange={searchChange}
            />
        </div>
    );
}

export default SearchBox;