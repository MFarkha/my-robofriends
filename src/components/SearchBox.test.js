import SearchBox from './SearchBox';
import React from 'react';
import { shallow } from 'enzyme';

it('expects to render the SearchBox component', () => {
    const mockSearchChange = jest.fn();
    const wrap = shallow(<SearchBox searchChange={mockSearchChange}/>);
    expect(wrap).toMatchSnapshot();
});