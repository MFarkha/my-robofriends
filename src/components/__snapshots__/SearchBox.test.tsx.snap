// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the SearchBox component 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="search robots"
    className="pa3 ba b--green bg-lightest-blue"
    name="searchBox"
    onChange={[MockFunction]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
